File: 2oo1s/BOJ_1343.py
# 10분
import sys
input = sys.stdin.readline

board = input()  # 보드판 입력

board = board.replace("XXXX", "AAAA")
board = board.replace("XX", "BB")

# 'AAAA'와 'BB' 폴리오미노로 변경을 해주고난 뒤에도
# X가 남아있다면,
# 폴리오미노로 다 덮을 수 없는 보드판이므로 -1 출력
if 'X' in board:
    print(-1)

else:
    print(board)

File: Austin-Choi/BOJ_1343.java
package com.edu.algo.test;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BOJ_1343 {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine(), " ");
		
		String str = st.nextToken();
		//문자열 "XXXX"를 우선으로 찾아서 "AAAA"로 치환
		str = str.replace("XXXX", "AAAA");
		str = str.replace("XX", "BB");
		
		
		//아직 남아있는 X는 치환되지 못한 문자열이 있는 것
		if(str.contains("X"))
			System.out.println(-1);
		else
			System.out.println(str);
	}
}
